nlb_service: add support for HTTPS health checking

falzm opened the pull request:
This change adds support for Network Load Balancer service HTTPS health
checking.

greut commented:
@pyr @vincentbernat adding the `context.Context` and killing `AsyncInfo` from egoscale simplifies everyone experience with it. go-cloudstack might steal it back 😜

greut commented:
this is the first step to be able to have a _per resource and action_ timeout.

https://www.terraform.io/docs/configuration/resources.html#timeouts

Commit:
nlb_service: add support for HTTPS health checking

This change adds support for Network Load Balancer service HTTPS health
checking.

Commit:
fixup! nlb_service: add support for HTTPS health checking

pierre-emmanuelJ reviewed (approved):
LGTM
